spec/index.bs

```diff
@@ -14028,6 +14029,22 @@ interface GPUCanvasContext {
             </div>
         </div>
 
+    : <dfn>getConfiguration()</dfn>
+    ::
+        Returns the context configuration.
+
+        <div algorithm=GPUCanvasContext.getConfiguration>
+            <div data-timeline=content>
+                **Called on:** {{GPUCanvasContext}} |this|.
+
+                **Returns:** {{GPUCanvasConfiguration}} or `null`
+
+                [=Content timeline=] steps:
+
+                1. Return |this|.{{GPUCanvasContext/[[configuration]]}}.
```

Comment:
Looking at the `configure()` algorithm, this `[[configuration]]` is the value that were passed into `configure()` without any changes, which leads to a couple of questions:

 - It seems like this should actually return a copy of `[[configuration]]` so that we don't imply that you could change the internal config by getting the dictionary and setting, say, a new format.
 - From the discussion on #4828 it sounded like part of the intent here was to feature detect what the system is actually using. Are there any scenarios where that may differ from what was passed in? (I know that was the case for WebGL sometimes, I don't recall here.) Or is there any non-trivial default resolution that we need to reflect. I'm not in a great position to research that at the moment but I can try to follow up on it in the next day or so if nobody else does the digging.

Otherwise this looks like what we want!

Reply:
Technically it's OK to just return `[[configuration]]`, but it would be clearer to read if it said "return a copy of".

(The reason it's OK is that getConfiguration() returns GPUCanvasConfiguration which is a dictionary, which gets created from scratch each time getConfiguration is called: https://webidl.spec.whatwg.org/#js-dictionary)

> * Are there any scenarios where that may differ from what was passed in?

This is kind of funny to specify because it exists specifically for feature detection in browsers which _don't_ currently match the spec. For example, in some browser which understands WebGPU but doesn't understand the `toneMapping` option, sites could expect it to either:
- not implement the `toneMapping` member at all so it will be missing from `getConfiguration()` - this is nonstandard but kind of like implementing an earlier version of the spec
- accept `toneMapping: { mode: 'extended' }` but reflect `toneMapping: { mode: 'standard' }` - this is totally nonstandard (as this PR stands), and I would not expect browsers to do it

If we want to standardize the behavior we need to make HDR WebGPU support optional. I think we don't want to; we want all browsers to handle HDR output from WebGPU and just tonemap it to SDR as a basic implementation.

Reply:
> I think we don't want to; we want all browsers to handle HDR output from WebGPU and just tonemap it to SDR as a basic implementation.

(Even if that tonemapping is just clamping - that's the same as `'standard'`; pages should look at media queries to determine hardware support.)

---

I guess to put what I'm saying in a more useful way, the thing we want to guarantee is that if the `dynamic-range` media query indicates HDR support AND `getConfiguration()` shows that the browser implements `toneMapping`, THEN you will get actual HDR support (not clamp to the SDR range of an HDR display).

Maybe that is something that we can say, non-normatively.

Reply:
Thank you both! I've addressed your concerns in latest commit.